download/download.go

```diff
@@ -405,6 +405,13 @@ func (d *Downloader) download(ctx context.Context, m metrics.Metrics) (*download
 			longPoll: d.longPollingEnabled,
 		}, nil
 	default:
+		if d.logger.GetLevel() == logging.Debug && resp.Body != nil {
+			body, err := io.ReadAll(resp.Body)
+			if err == nil {
```

Comment:
If we fail to fetch the body, perhaps we should log that too.

Reply:
I was thinking about that too. But I noticed we follow a similar pattern for the AWS downloader so thought to keep it as-is and we can update if needed.